### PR TITLE
TM-78: Get Entity REST Operation

### PR DESCRIPTION
**Pull request by janoliver20**
- Added getEntity and getEntities to entity service.
- Implemented REST methods
- Changed dbo models a bit

### CODE REVIEW

File: be/src/controllers/rest.router.ts [hunk 1]
```diff
@@ -88,9 +104,28 @@ router.delete('/vocab/:id',  (req: Request, res: Response, next: NextFunction) =
...
+    if (!checkQueryParams(['text', 'type', 'createdSince', 'modifiedSince', 'sort', 'offset', 'rows'], req?.query)) {
+      next(new KnowledgeError(400, 'Bad Request', 'Invalid query parameters'));
+    } else {
+      const queryListModel: ListQueryModel = {
+        ...req?.query,
+        modifiedSince: !!req?.query?.modifiedSince ? new Date(`${req?.query.modifiedSince}`) : undefined,
+        createdSince: !!req?.query?.createdSince ? new Date(`${req?.query.createdSince}`) : undefined
+      };
+      const vocabID = req.params.vocabID;
+      entityServiceInstance.getEntities(vocabID, queryListModel, req.params.id)
```
Comment (ja-fra) [inline be/src/controllers/rest.router.ts]: Why `req.params.id` here as parameter?

File: be/src/services/entity.service.ts [hunk 2]
```diff
@@ -19,16 +21,20 @@ export class EntityService {
...
     }
 
-    // TODO remove rule when implemented
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    getEntity(vocabID: string, entityID: string): Promise<unknown> {
-        throw new KnowledgeError(501, 'Not Implemented', 'Operation not implemented',
-          {id:'dummyEntityErrorPayload'}
-        );
+    getEntity(vocabID: string | ObjectId, entityID: string | ObjectId): Promise<Entity> {
+        return this.collection.findOne({_id: new ObjectId(entityID), vocabulary: new ObjectId(vocabID)}).then(x => <Entity>x);
```
Comment (ja-fra) [inline be/src/services/entity.service.ts]: Please, also check that the provided `vocabularyId` exists as a Vocabulary in the DB.

**Review by ja-fra — CHANGES_REQUESTED** [review]

File: be/src/services/entity.service.ts (hunk 2 above)
Comment (ManuelTS) [reply be/src/services/entity.service.ts]: GET entity + entities works, search functionality of lists is excluded and will be programmed in another branch

File: be/src/controllers/rest.router.ts [hunk 3]
```diff
@@ -129,9 +129,32 @@ router.delete('/vocab/:id', (req: Request, res: Response, next: NextFunction) =>
         .catch(next);
 });
 
-router.get('/vocab/:id/entities', (req: Request, res: Response, next: NextFunction) => {
+router.get('/vocab/:vId/entities', (req: Request, res: Response, next: NextFunction) => {
```
Comment (dan0bar) [inline be/src/controllers/rest.router.ts:132]: Not sure if we need `list Entities` in this PR

File: be/src/services/entity.service.ts [hunk 4]
```diff
@@ -1,98 +1,117 @@
...
-              if(!!result?._id) {
-                return result as Entity;
-              }
-              throw new KnowledgeError(404,
-                'Not found',
-                `Target entity with id '${entityID}' in vocabulary '${vocabID}' not found`);
-          })
-    }
+  public getEntities(vocabID: string | ObjectId): Promise<Entity[]> {
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (dan0bar) [inline be/src/services/entity.service.ts:42]: Also should be in another PR

**Review by dan0bar — APPROVED** [review]

File: be/src/controllers/rest.router.ts (hunk 3 above)
Comment (ManuelTS) [reply be/src/controllers/rest.router.ts:132]: The ticket just states reading entities, that is why I included GET entitiy and GET entities, but the latter without its search functionality, which will be done in an extra branch

**Review by ManuelTS — APPROVED** [review]
